mcp/src/main/java/io/modelcontextprotocol/spec/McpSession.java

```diff
@@ -63,10 +63,10 @@ default Mono<Void> sendNotification(String method) {
 	 * parameters with the notification.
 	 * </p>
 	 * @param method the name of the notification method to be sent to the counterparty
-	 * @param params a map of parameters to be sent with the notification
+	 * @param params parameters to be sent with the notification
 	 * @return a Mono that completes when the notification has been sent
 	 */
-	Mono<Void> sendNotification(String method, Map<String, Object> params);
+	Mono<Void> sendNotification(String method, Object params);
```

Comment:
Let's keep in mind to take note of the breaking change in the release docs. I think it's worth the effort though.